### ISSUE TITLE
Suggest missing function arguments

### ISSUE THREAD

**Issue by kittykatattack** [elm-lang/error-message-catalog#119]
Consider the following:
Here's function that displays a paragraph based on some parameters:

```
paragraphView paragraphId =
  let

    answeredQuestions =
      List.filter (\question -> not (String.isEmpty question.answer)) model.questions

    sentencesBelongingToParagraph paragraphId =
       List.filter (\question -> question.paragraphId == paragraphId ) answeredQuestions
  in
  p [] (List.map sentenceView (sentencesBelongingToParagraph paragraphId))
```

However, it requires access to `model`, which is not accessible in its current scope. If a user tries to call this function, the following error message is generated:

```
Cannot find variable `model`

261|       List.filter (\question -> not (String.isEmpty question.answer)) model.questions
                                                                           ^^^^^
Maybe you want one of the following?

    Model
    code
    del
    node
```

The actual problem isn't detected. Maybe we need to add something like this:

```
"Or, perhaps you need to include `model` as an argument to `paragraphView`?
```

**Comment by ckoster22** [comment]
`model` could be a constant at a higher scope. For example, it could be a constant declared at the beginning of your module.. a sibling to the paragraphView declaration.

With this in mind the compiler error makes perfect sense. It's looking for a variable called `model` which could be defined at a higher scope than the function's inner scope. There's no requirement that variables used within functions must be passed into that function.

**Comment by sudhirvkumar** [comment]
@kittykatattack  @ckoster22 I agree that all variables used within functions need not be passed as arguments.
If a variable is not in scope, then it has to be passed as a parameter and there is no other way to make it available to the function, so I believe adding this message will help!
